dfc/tests/integration_test.go

```diff
@@ -0,0 +1,301 @@
+package dfc_test
+
+import (
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/NVIDIA/dfcpub/dfc"
+	"github.com/NVIDIA/dfcpub/pkg/client"
+)
+
+type repFile struct {
+	repetitions int
+	filename    string
+}
+
+type metadata struct {
```

Comment:
in general, reduce scope of variables are better, when i see a big struct like this, i will see if they are all necessary to put into one struct, some can be local.
i did see 'm' is passed into two calls, even if they share most of the fields, still can be separated into sub struct.
i am ok with this as this is a test code, just some thoughts.